src/screens/UploadFiles.tsx

```diff
@@ -40,11 +78,89 @@ function UploadFiles() {
         return {
           ...state,
           file: action.value,
+          fileValidationStatus: null,
```

Comment:
I feel like there's a lib that will take care of this for you.  Maybe immer or react final form.

Reply:
I could use react-hook-form, but it would require a big overhaul to fit into that libraries system. With the validations being so simple (required or not) I'm not sure it is worth it? Thoughts?